enhancements/scheduling/prevent-user-workloads-from-being-scheduled-on-control-plane-nodes.md

```diff
@@ -0,0 +1,702 @@
+
+### Expected Outcomes
+
+- Improved protection of critical cluster resources, ensuring that control plane nodes <!--and nodes with 
+ resources --> are shielded from inappropriate workloads.
+- Increased flexibility for administrators to tailor scheduling policies to the specific needs of their organization and operational environment.
+
+
+
+### Workflow Description
+
+<!-- Explain how the user will use the feature. Be detailed and explicit.
+Describe all of the actors, their roles, and the APIs or interfaces
+involved. Define a starting state and then list the steps that the
+user would need to go through to trigger the feature described in the
+enhancement. Optionally add a
+[mermaid](https://github.com/mermaid-js/mermaid#readme) sequence
+diagram.
+
+Use sub-sections to explain variations, such as for error handling,
+failure recovery, or alternative outcomes.
+
+For example:
+
+**cluster creator** is a human user responsible for deploying a
+cluster.
+
+**application administrator** is a human user responsible for
+deploying an application in a cluster.
+
+1. The cluster creator sits down at their keyboard...
+2. ...
+3. The cluster creator sees that their cluster is ready to receive
+   applications, and gives the application administrator their
+   credentials. -->
+   
+This section outlines the workflow for enforcing the proposed scheduling policies that prevent non-control plane workloads from being scheduled on control plane nodes and allow for exceptions based on administrative configurations.
+
+#### Actors
+
+- Cluster Administrator: Responsible for configuring scheduling policies, managing exceptions, and overseeing cluster operations.
+- User: Individuals or services attempting to deploy workloads within the cluster.
+- Admission Controller: The mechanism that intercepts pod scheduling requests to enforce scheduling policies.
+
+#### Workflow Steps
+
+1. Policy Configuration:
+
+- The cluster administrator defines and configures the scheduling policies using a new or extended configuration resource within the OpenShift API. 
+- The admission plugin configuration will be stored under KAO [configuration](https://github.com/openshift/cluster-kube-apiserver-operator/blob/189ce0e7d47864366c741d3d5b9bd9c820421c14/bindata/assets/config/defaultconfig.yaml#L4C3-L4C15). The configuration will not be writable in the first iteration. There's going to be a predefined list of authorized SAs/users for all openshift components allowed to run on control plane nodes. Admins will be allowed to extend the authorized list of SAs/users only through config.openshift.io/v1/scheduler singleton object.
+- The policy configuration will be TechPreviewNoUpgrade feature gated. The feature gate name needs to be `OpenShift` prefixed.
+
+2. Pod Scheduling Request:
+
+- A user creates a new pod or updates an existing pod specification, including scheduling preferences such as node selectors or tolerations.
```

Comment:
TODO: We need to mention we strongly recommend to avoid tolerating "[node-role.kubernetes.io/master:NoSchedule](http://node-role.kubernetes.io/master:NoSchedule)" taint as a precaution against admission rejection. I.e. as a good practice. Under normal circumstances a workload does not toleration the taint. However, users might accidentally copy-paste a manifest and alter it while forgetting to remove the toleration for the taint. Something to be mentioned in 4.16 release notes and in the admission plugin documentation.

Should this become text somewhere in the proposal or is this just a reminder for us when we are dealing with release notes and docs?

Reply:
Good practice for users. OpenShift documentation is more suitable for this. Plus release notes.